### DISCUSSION TITLE
Set Key In "values" Object To Custom Value

### DISCUSSION THREAD

**Discussion by ApplePieGiraffe** [react-hook-form/react-hook-form#5519] [Q&A] [answered]
I'm trying to use `useController` to hook up an input element with `type="file"` to `react-hook-form`. I'd like to convert an uploaded image to a string and send it as part of the object containing other values in the same form. I can't seem to be able to set the value of the input element (because React won't let you do so for `<input type="file">`) and `onChange` simply passes the value of the input element to `react-hook-form` (which, at the moment, is just the default value I set—an empty string). 

Is there any way I can set the value of the corresponding key in the values object returned by `react-hook-form` to something other than the value of the input element it is registered with? Like, in this case, I want to set the value of the `image` property to a converted string of the image, not the empty string that is the actual value of the `<input type="file">` element. If there is, how can I do that?

**Comment by bluebill1049** [comment, marked as answer]
https://codesandbox.io/s/long-sun-nsfbk?file=/src/App.js

```tsx
import * as React from "react";
import { useController, useForm } from "react-hook-form";
import Headers from "./Header";
import "./styles.css";

let renderCount = 0;

const FileInput = ({ control, name }) => {
  const { field } = useController({ control, name });
  const [value, setValue] = React.useState("");
  return (
    <input
      type="file"
      value={value}
      onChange={(e) => {
        setValue(e.target.value);
        field.onChange(e.target.files);
      }}
    />
  );
};

export default function App() {
  const { register, handleSubmit, control } = useForm();
  const onSubmit = (data) => console.log(data);
  renderCount++;

  return (
    <div>
      <Headers
        renderCount={renderCount}
        description="Performant, flexible and extensible forms with easy-to-use validation."
      />

      <form onSubmit={handleSubmit(onSubmit)}>
        <input {...register("firstName")} placeholder="First Name" />
        <FileInput name="file" control={control} />
        <input type="submit" />
      </form>
    </div>
  );
}
```

is this helpful?

**Reply by ApplePieGiraffe** [reply]
Well, the example successfully sets the value of the input element to a `FileList` object, but in my case I would like to convert the file to a string first and then set that string to the value of the `file` key in the `values` object. So instead of,
```
{ firstName: '', file: FileList } 
```
I'd have something like,
```
{ firstName: '', file: <string> }
```
So I'm wondering if there's a way to set the value of a key in that object to something other than the value of the input element it is registered to (in this case, a string instead of the `FileList` object). Or maybe that's not possible/the intention of `react-hook-form`, I'm not sure.

**Reply by bluebill1049** [reply]
you can build whatever you want in your controlled component, hook form's job is only to collect and validate the data.

**Reply by ApplePieGiraffe** [reply]
Okay. I think I'll just create a new object from the `values` object `react-hook-form` returns, then (since I cannot modify it directly). Thanks for taking a look at this. 🙂
